src/models/tools/geometry/geometry-content.ts

```diff
@@ -821,3 +833,153 @@ export const GeometryContentModel = types
   });
 
 export type GeometryContentModelType = Instance<typeof GeometryContentModel>;
+
+interface IBoardImportProps {
+  axisMin?: JXGCoordPair;
+  [prop: string]: any;
+}
+interface IBoardImportSpec {
+  properties?: IBoardImportProps;
+}
+
+interface IPointImportSpec {
+  type: "point";
+  parents: [number, number];
+  properties?: any;
+}
+
+interface IVertexImportSpec extends IPointImportSpec {
+  angleLabel?: boolean;
+}
+
+interface IPolygonImportSpec {
+  type: "polygon";
+  parents: IVertexImportSpec[];
+  properties?: any;
+}
+
+interface IImageImportSpec {
+  type: "image";
+  parents: {
+    url: string;
+    coords: JXGCoordPair;
+    size: JXGCoordPair;
+  };
+  properties?: any;
+}
+
+interface IMovableLineSpec {
+  type: "movableLine";
+  parents: [JXGCoordPair, JXGCoordPair];
+  properties?: any;
+}
+
+type IObjectImportSpec = IPointImportSpec | IPolygonImportSpec | IImageImportSpec | IMovableLineSpec;
+
+function preprocessImportFormat(snapshot: any) {
+  const boardSpecs = snapshot.board as IBoardImportSpec;
+  const objectSpecs = snapshot.objects as IObjectImportSpec[];
+  if (!objectSpecs) return snapshot;
+
+  function addBoard(boardSpec: IBoardImportSpec) {
+    const { properties } = boardSpec || {} as IBoardImportSpec;
+    const { axisMin, ...others } = properties || {} as IBoardImportProps;
+    const boundingBox = getBoardBounds(axisMin);
+    changes.push(defaultGeometryBoardChange({ boundingBox, ...others }));
+  }
+
+  const changes: JXGChange[] = [];
+  addBoard(boardSpecs);
+
+  // map import names to internal names
+  const kPropsMap: { [p: string]: string } = {
```

Comment:
Could leave this responsibility up to the change handler

Reply:
After we talked about it I implemented this suggestion and like the way it turned out. It removes `clientFillColor`, etc. from the API and encapsulates it as an implementation detail. Good idea.